packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -1066,7 +1070,14 @@ export class GraphiQL extends React.Component<GraphiQLProps, GraphiQLState> {
         );
       }
 
-      const totalResponse: FetcherResult = { data: {} };
+      const isIncrementalDelivery = (
+        result: FetcherResult,
+      ): result is IncrementalDeliveryResult => {
+        // @ts-ignore
```

Comment:
again, no `@ts-ignore` needed here

Reply:
again, here's the code I had, no ts ignore needed:

```ts

 const totalResponse: FetcherResult = { data: {} };

      // _fetchQuery may return a subscription.
      const subscription = await this._fetchQuery(
        editedQuery as string,
        variables as string,
        headers as string,
        operationName as string,
        shouldPersistHeaders as boolean,
        (result: FetcherResult) => {
          if (queryID === this._editorQueryID) {
            if (typeof result !== 'string' && result.hasNext) {
              if (result.errors) {
                totalResponse.errors = [
                  ...(totalResponse?.errors || []),
                  ...result?.errors,
                ];
              }
              if ('path' in result) {
                const [path, index] = result.path;
                const data = totalResponse?.data[path] || [];
                // place them at the exact index. this matters a lot
                data[index] = result.data;
                totalResponse.data = { ...totalResponse?.data, [path]: data };
              } else {
                totalResponse.data = {
                  ...totalResponse?.data,
                  ...result?.data,
                };
              }
            }
            this.setState({
              isWaitingForResponse: false,
              response: GraphiQL.formatResult(totalResponse),
            });
          } else {
            this.setState({
              isWaitingForResponse: false,
              response: GraphiQL.formatResult(result),
            });
          }
        },
      );

      this.setState({ subscription });
      
  ```
and 
```ts
export type FetcherResult =
  | {
      data: IntrospectionQuery;
      errors?: Array<any>;
      hasNext?: boolean
    }
  | string
  | { data?: any; errors?: Array<any>; hasNext?: boolean }
  // for IncrementalDelivery via mereos
  | {
      data: any;
      errors?: any[];
      path: [string, number];
      hasNext: boolean;
    };

```

Reply:
the type gaurd is a nice pattern in theory but something is just not right if you need these ignores. if you try bringing that condition i wrote back inline, you'll see the inference engine picks it up, or it should at least?

Reply:
Ya, I'll give it another check. For some reason I think the language server in vscode didn't complain, but did in IntelliJ.

I'll confirm, but when I did the type narrowing to object (instead of not string), it didn't like that "hasNext" was likely not in the object, seeing as the union finds the lowest common denominator. So in the typeguard I think it's fine. 

The rest should be removed though. 

Cheers! 🥂